src/org/skypro/skyshop/Search/SearchEngine.java

```diff
@@ -1,53 +1,52 @@
 package org.skypro.skyshop.Search;
 
+import java.util.ArrayList;
+import java.util.List;
+
 public class SearchEngine {
-    Searchable [] searchables;
-    int count;
+    List<Searchable> searchables;
```

Comment:
Не забывай про инкапсуляцию
```suggestion
    private List<Searchable> searchables;
```